libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -417,6 +417,40 @@ void AP_DDS_Client::update_topic(geographic_msgs_msg_GeoPoseStamped& msg)
     }
 }
 
+void AP_DDS_Client::update_topic(sensor_msgs_msg_Imu& msg)
+{
+    update_topic(msg.header.stamp);
+    strcpy(msg.header.frame_id, BASE_LINK_FRAME_ID);
+
+    auto &imu = AP::ins();
+    auto &ahrs = AP::ahrs();
+
+    WITH_SEMAPHORE(ahrs.get_semaphore());
+
+    Quaternion orientation;
+    if (ahrs.get_quaternion(orientation)) {
+        msg.orientation.x = orientation[0];
+        msg.orientation.y = orientation[1];
+        msg.orientation.z = orientation[2];
+        msg.orientation.w = orientation[3];
+    }
+    msg.orientation_covariance[0] = -1;
+
+    Vector3f accel_data = imu.get_accel();
+    Vector3f gyro_data = imu.get_gyro();
+
+    // Populate the message fields
+    msg.linear_acceleration.x = accel_data.x;
```

Comment:
@Ryanf55 do we need FRD to RFU conversion here?

Reply:
We should transform to ENU if this is compliant with REP-147, otherwise this needs a wiki entry to explain the divergence.

Reply:
OR, NED if desired. Commented above: https://github.com/ArduPilot/ardupilot/pull/26187/files#r1495230958

REP-103 suffix frames allow NED if desired.
https://www.ros.org/reps/rep-0103.html#suffix-frames

Reply:
@Astik-2002  This needs to be resolved and documented before merge. Have you checked what coordinate system  `AP_InertialSensor::get_accel` is in?

Reply:
The default coordinate system is NED